### PR TITLE
17759: Adds 'features' key to react details which specifies a subset of features to compute metrics for

### PR DESCRIPTION
**Pull request by cademack**
Adds 'features' key to react details. This key takes a list of feature names and uses that to select the subset of features that metrics like feature_residuals, MDA, contributions, etc will be computed for.

This problem is proving to be quite difficult for the robust version of these metrics, as it is far from well-suited to compute for a subset of the features. It isn't really a problem for the full version of these metrics.

It might be the best balance of dev-time vs functionality to make this only save compute in FULL workflows, but rather only return the requested feature metrics when doing things in ROBUST?

### CODE REVIEW

File: unit_tests/ut_h_react_explain.amlg [hunk 1]
```diff
@@ -754,6 +754,31 @@
...
+				context_values (list 6.9 7 7 6)
+				details (assoc
+					"robust_computation" (true)
+					"case_feature_residuals" (true)
+					"features" (list "A" "C")
+				)
+			))
+	))
+
+	(print "Robust case residuals and residual convictions for a prediction: ")
```
Comment (howsoRes) [inline unit_tests/ut_h_react_explain.amlg]: for a prediction -> for specific features

File: unit_tests/ut_h_react_explain.amlg [hunk 2]
```diff
@@ -823,6 +848,49 @@
...
+					"robust_computation" (true)
+					"case_feature_residuals" (true)
+					"local_case_feature_residual_convictions" (true)
+					"global_case_feature_residual_convictions" (true)
+					"features" (list "A" "B")
+				)
+			))
+	))
+
+	(print "robust local - similar to previous: ")
```
Comment (howsoRes) [inline unit_tests/ut_h_react_explain.amlg]: specific features robust local - similar to previous:

File: unit_tests/ut_h_react_explain.amlg [hunk 3]
```diff
@@ -823,6 +848,49 @@
...
+				"A" 4
+				"B" .7
+		)
+		thresh
+			(assoc
+				"A" 4
+				"B" .7
+			)
+	))
+	(print "robust global has values: ")
```
Comment (howsoRes) [inline unit_tests/ut_h_react_explain.amlg]: specific feature robust global has values:

**Review by howsoRes — APPROVED** [review]

File: unit_tests/ut_h_react_explain.amlg (hunk 3 above)
Comment (cademack) [reply unit_tests/ut_h_react_explain.amlg]: I already fixed this one, I think you are viewing an older commit

**Pull request merged**